lib/diaspora/webfinger.rb

```diff
@@ -0,0 +1,125 @@
+#   Copyright (c) 2010-2012, Diaspora Inc.  This file is
+#   licensed under the Affero General Public License version 3 or later.  See
+#   the COPYRIGHT file.
+
+module Diaspora
+  class Webfinger
+    include Diaspora::Logging
+
+    attr_accessor :host_meta_xrd, :webfinger_profile_xrd,
+                  :webfinger_profile, :hcard, :hcard_xrd, :person,
+                  :account, :ssl
+
+    def initialize(account)
+      self.account = account
+      self.ssl = true
+    end
+
+
+    def fetch
+      return person if existing_person_with_profile?
+      create_or_update_person_from_webfinger_profile!
+    end
+
+    def self.in_background(account, opts={})
+      Workers::FetchWebfinger.perform_async(account)
+    end
+
+    #everything below should be private I guess
+    def account=(str)
+      @account = str.strip.gsub('acct:','').to_s.downcase
```

Comment:
Space missing after comma.<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.